diff --git a/hp-wmi.c b/hp-wmi.c
index 0fd045d..3652141 100644
--- a/hp-wmi.c
+++ b/hp-wmi.c
@@ -529,7 +529,7 @@ static int hp_wmi_get_fan_count_userdefine_trigger(void)
 static void hp_victus_s_work_handler(struct work_struct *work)
 {
     hp_wmi_get_fan_count_userdefine_trigger();
-    schedule_delayed_work(to_delayed_work(work), secs_to_jiffies(HP_VICTUS_S_THERMAL_PROFILE_TIMER_SECONDS));
+    schedule_delayed_work(to_delayed_work(work), (HP_VICTUS_S_THERMAL_PROFILE_TIMER_SECONDS * HZ));
 }
 
 
@@ -2691,7 +2691,7 @@ static int __init hp_wmi_init(void)
 			goto err_unregister_device;
 
         INIT_DELAYED_WORK(&hp_fan_control.victus_s_thermal_profile_trigger_work, hp_victus_s_work_handler);
-        schedule_delayed_work(&hp_fan_control.victus_s_thermal_profile_trigger_work, secs_to_jiffies(HP_VICTUS_S_THERMAL_PROFILE_TIMER_SECONDS));
+        schedule_delayed_work(&hp_fan_control.victus_s_thermal_profile_trigger_work, (HP_VICTUS_S_THERMAL_PROFILE_TIMER_SECONDS * HZ));
 	}
 
 	return 0;
@@ -2727,4 +2727,4 @@ static void __exit hp_wmi_exit(void)
 		platform_driver_unregister(&hp_wmi_driver);
 	}
 }
-module_exit(hp_wmi_exit);
\ No newline at end of file
+module_exit(hp_wmi_exit);
